apps/prairielearn/src/components/NavbarContext.tsx

```diff
@@ -17,9 +17,10 @@ export function ContextNavigation({
   if (!navPage) return '';
 
   const navPagesTabs = getNavPageTabs(resLocals.has_enhanced_navigation);
-  const navPageTabs = navPagesTabs[navPage];
+  const navPageTabs = navPagesTabs[navPage as keyof typeof navPagesTabs];
```

Comment:
I really don't like these type assertions because I think they are effectively removing any benefits of strict typing. Can we instead use zod to `safeParse` the `navPage` value and return the empty string if it doesn't parse? I know this will require more work to set up the correct zod schema for `navPagesTabs` but I feel like it's the right thing to do.